explainers/signed-exchange-subresource-subtitution-explainer.md

```diff
@@ -0,0 +1,455 @@
+may discard the entry if it is expired.
+
+## Content negotiation using Variants and Variant-Key
+
+The **allowed-alt-sxg** link headers can have
+[variants and variant-key](https://httpwg.org/http-extensions/draft-ietf-httpbis-variants.html)
+attributes to support content negotiation (eg:
+[WebP support](https://developers.google.com/speed/webp/faq#server-side_content_negotiation_via_accept_headers)).
+```
+Link: <https://publisher.example/image>;
+      rel="allowed-alt-sxg";
+      variants-05="accept;image/jpeg;image/webp";
+      variant-key-05="image/jpeg";
+      header-integrity="sha256-AAA"
+Link: <https://publisher.example/image>;
+      rel="allowed-alt-sxg";
+      variants-05="accept;image/jpeg;image/webp";
+      variant-key-05="image/webp";
+      header-integrity="sha256-BBB"
+Link: <https://publisher.example/image>; rel=preload; as=image;
+```
+If a UA supports WebP format, the UA can use the signed exchange which
+header-integrity is sha256-BBB if available in the cache. Otherwise the UA can
+use the signed exchange which header-integrity is sha256-AAA if available in the
+cache.
+
+When the sha256-AAA signed exchange exists in the cache but the sha256-BBB
+signed exchange doesn’t exists, the UA which supports WebP format MUST ignore
+the sha256-AAA signed exchange and fetch the original URL. Otherwise this can be
+used for sending tracking ID.
+
+The **alternate** link also can have
+[variants and variant-key](https://httpwg.org/http-extensions/draft-ietf-httpbis-variants.html)
+attributes to support content negotiation for recursive prefetch.
+
+- In **outer** HTTP response of article.html.sxg from feed.example:
+    ```
+    Link: <https://feed.example/publisher.example/image_jpeg.sxg>;
+          rel="alternate";
+          type="application/signed-exchange;v=b3";
+          variants-05="accept;image/jpeg;image/webp";
+          variant-key-05="image/jpeg";
+          anchor="https://publisher.example/image";
+    Link: <https://feed.example/publisher.example/image_webp.sxg>;
+          rel="alternate";
+          type="application/signed-exchange;v=b3";
+          variants-05="accept;image/jpeg;image/webp";
+          variant-key-05="image/webp";
+          anchor="https://publisher.example/image";
+    ```
+- In **inner** response header of article.html.sxg:
+    ```
+    Link: <https://publisher.example/image>;
+          rel="allowed-alt-sxg";
+          variants-05="accept;image/jpeg;image/webp";
+          variant-key-05="image/jpeg";
+          header-integrity="sha256-AAA"
+    Link: <https://publisher.example/image>;
+          rel="allowed-alt-sxg";
+          variants-05="accept;image/jpeg;image/webp";
+          variant-key-05="image/webp";
+          header-integrity="sha256-BBB"
+    Link: <https://publisher.example/image>; rel=preload; as=image;
+    ```
+If a UA supports WebP, the UA should prefetch **image_webp.sxg** which content
```

Comment:
```suggestion
If a UA supports WebP, the UA must prefetch **image_webp.sxg** holding a WebP image.
```

This isn't "should", since it's defined by the variants algorithm.